certbot-apache/certbot_apache/constants.py

```diff
@@ -62,9 +62,9 @@
     vhost_files="*.conf",
     logs_root="/var/log/apache2",
     version_cmd=['/usr/sbin/apache2', '-v'],
-    define_cmd=['apache2ctl', 'virtualhosts'],
-    restart_cmd=['apache2ctl', 'graceful'],
-    conftest_cmd=['apache2ctl', 'configtest'],
+    define_cmd=['/usr/sbin/apache2', '-S'],
```

Comment:
The reason for the different output is that `apache2ctl` additionally sources `/etc/conf.d/apache2` and then adds the `APACHE2_OPTS` defined there to the `apache2` call:

Example for `APACHE2_OPTS` in `/etc/conf.d/apache2`:
`APACHE2_OPTS="-D INFO -D LANGUAGE -D SSL -D PHP -D SECURITY -D PROXY"`

Function that gets executed when calling `apache2ctl virtualhosts`:
```
virtualhosts() {
        checkconfig || return 1
        ${APACHE2} ${APACHE2_OPTS} -S
}
```
The options in `APACHE2_OPTS` are used in Gentoo to define includes in vhosts, like in `/etc/apache2/vhosts.d/00_default_ssl_vhost.conf`:
```
<IfDefine SSL>
<IfDefine SSL_DEFAULT_VHOST>
<IfModule ssl_module>

# see bug #178966 why this is in here

# When we also provide SSL we have to listen to the HTTPS port
# Note: Configurations that use IPv6 but not IPv4-mapped addresses need two
# Listen directives: "Listen [::]:443" and "Listen 0.0.0.0:443"
Listen 443

<VirtualHost _default_:443>
        ...
</VirtualHost>
</IfModule>
</IfDefine>
</IfDefine>
```
For me, it worked perfectly without having the flags included but there may be indeed some side effects. But while using `apache2ctl` renders certbot completely useless for Gentoo systemd users (automatic installation AND certonly are broken), using `apache2` "only" breaks the automatic installation, leaving workarounds using certonly functional.

To fix this issue completely, certbot requires support for different commands depending on the init system. Until that is done I would say generally enabling all users to use certonly is better then breaking the support for one group (especially since systemd seems to be becoming industry standard) while enabling full support for the other.